lib/reek/tree_walker.rb

```diff
@@ -6,12 +6,15 @@
 
 module Reek
   #
-  # Traverses a Sexp abstract syntax tree and fires events whenever
-  # it encounters specific node types.
+  # Traverses an abstract syntax tree and fires events whenever it encounters
+  # specific node types.
   #
   # SMELL: This class is responsible for counting statements and for feeding
   # each context to the smell repository.
   #
+  # TODO: Make TreeWalker responsible only for creating Context objects, and
```

Comment:
Ok, but we'll still have:

1.) Traverse AST to adorn it with sexp extensions
2.) Traverse it again to generate a tree of contexts
3.) Traverse the context tree

Right?